src/components/furniture/HardwareHandle.vue

```diff
@@ -2,7 +2,7 @@
   <div
```

Comment:
In this component, there are a few anti-pattern things that make it a bit "spagethi".

We want to keep things separated where they belong and try not to mix things up.

So first, we see a lot of CSS mixed inside the javascript functions. 

# Step 1: extract CSS and keep it separated

Let's take that out ant create new css classes in our <style> </style> tag. 

There we can define all the css that does not need any computation logic. That will be the "default" css applied to out template. Then we can, "a la marge", over-write this styling:

Result step 1:

Here is our "base" styling: 

```vue
<template>
  <div
    v-if="space.hardware === 'door'"
    class="handle door"
  ></div>
  <div
    v-if="space.hardware === 'foldingDoor'"
    class="handle foldingDoor"
  ></div>
  <div v-if="space.hardware === 'drawer'" class="handle drawer"></div>
</template>

<script>
 // your code here
</script>

<style scoped>
.door {
  marginright: 5%;
}
.handle {
  height: 12px;
  width: 12px;
  border: solid;
  background-color: white;
  border-radius: 50%;
}

.drawer {
  float: 'left';
  marginleft: 50%;
  margintop: 5%;
  transform: translateX(-50%);
}

.foldingDoor {
  float: bottom;
  marginleft: 50%;
  transform: translateX(-50%);
}
</style>

```

Reply:
# Step 2: custom inline css with computed values

Let's now apply the logic that CANNOT be defined only with css. All the style applied directly via the `style={}` will always overwrite the style defined in classes. That is because of CSS rules of precedence (and the reason it's called Cascading styling sheet. [Check this image](https://pbs.twimg.com/media/EdtOnvXXsAErUiL?format=jpg&name=large) if you don't know what i'm talking about).

So now we should have a template like that right?

```vue
<template>
  <div
    v-if="space.hardware === 'door'"
    class="handle door"
    :style="{ float: this.doorSide }"
  ></div>
  <div
    v-if="space.hardware === 'foldingDoor'"
    class="handle foldingDoor"
    :style="{ marginTop: this.folfingDoorMarginTop }"
  ></div>
  <div v-if="space.hardware === 'drawer'" class="handle drawer"></div>
</template>

// Here below we should compute those values

```

Reply:
# Step 3: make these specific computed values

So the first one is dooSide. And it should return 'left' or 'right'. How do we know doorSide? It comed from the door if. 

So let's go 

```vue
<script>
  //...
  computed: {
    ...mapGetters(['shelfs']),
    doorSide() {
      return this.space.id === 0 ? 'left' : 'right';
    },
   }
  //...
</script>

```

Easy.

In the original version, you call a function to define the value and store it inside the data object. This not correct and not necessary (you don't need to store that value and you should not call side effects inside a computed value function.


Okay now `this.folfingDoorMarginTop`

So basically, if it's in high position, it should be "something" and if it's in low position it's 5%.

```vue
<script>

  //...
  computed: {
    //...
    foldingDoorMarginTop() {
      return this.isDoorInHighPosition
        ? this.shelf.height * 3 * 0.8 + 'px'
        : '5%';
    },
   }
  //...
  
</script>
```

with `isDoorInHighPosition`: 

```vue
<script>

  //...
  computed: {
    //...
    isDoorInHighPosition() {
      return this.doorHeight > 150;
    }, 
  }
  //...
</script>

```

Very clean.

The only missing now is this.doorHeight

```vue
<script>

  //...
  computed: {
    //...
    doorHeight() {
      // this is pretty hard to understand. Could be improved. Try to avoid variables like `arr` or `string` because it doesn't specify what is really is. 
      return this.shelfs.reduce(
        (arr, shelf) => (this.shelf.id < shelf.id ? arr + shelf.height : arr),
        0
      );
    },
  }
  //...
</script>

```

Reply:
# Full component (aproximative, please check that it works)

Should look like

```vue
<template>
  <div
    v-if="space.hardware === 'door'"
    class="handle door"
    :style="{ float: this.doorSide }"
  ></div>
  <div
    v-if="space.hardware === 'foldingDoor'"
    class="handle foldingDoor"
    :style="{ marginTop: this.foldingDoorMarginTop }"
  ></div>
  <div v-if="space.hardware === 'drawer'" class="handle drawer"></div>
</template>

<script>
import { mapGetters } from 'vuex';
export default {
  props: ['space', 'shelf'],
  computed: {
    ...mapGetters(['shelfs']),
    doorSide() {
      return this.space.id === 0 ? 'left' : 'right';
    },
    foldingDoorMarginTop() {
      return this.isDoorInHighPosition
        ? this.shelf.height * 3 * 0.8 + 'px'
        : '5%';
    },
    isDoorInHighPosition() {
      return this.doorHeight > 150;
    },
    doorHeight() {
      // this is pretty hard to understand. Could be improved
      return this.shelfs.reduce(
        (arr, shelf) => (this.shelf.id < shelf.id ? arr + shelf.height : arr),
        0
      );
    },
  },
};
</script>

<style scoped>
.door {
  marginright: 5%;
}
.handle {
  height: 12px;
  width: 12px;
  border: solid;
  background-color: white;
  border-radius: 50%;
}

.drawer {
  float: 'left';
  marginleft: 50%;
  margintop: 5%;
  transform: translateX(-50%);
}

.foldingDoor {
  float: bottom;
  marginleft: 50%;
  transform: translateX(-50%);
}
</style>

```